### ISSUE TITLE
Add support for remote/network locations for Local Files music provider

### ISSUE THREAD

**Issue by marcelveldt** [music-assistant/hass-music-assistant#71]
The local files integration now only supports local directories (e.g. /media/music). It would be nice if remote locations can also be supported.

- Network location (SMB/CIFS)
- Google Drive / Onedrive ?
- ??

**Comment by jeroenterheerdt** [comment]
I came here looking for this, so +1! :)

**Comment by vk2him** [comment]
@marcelveldt  - I have the files from my network mounted Synology NAS added to my /media directory, so this integration sees those files and can play them. Add/modify the following in your configuration.yaml
Note:
Replace  //192.168.1.252/music with the network path to your files
The above path will be mounted to the following directory (change as needed) /media/nasfiles

```
# For mounting Network media
shell_command:
  mount_nas_folder: mkdir -p /media/nasfiles;mount -t cifs -o vers=3.0,noserverino,username=nasusername,password=naspassword,domain=WORKGROUP //192.168.1.252/music /media/nasfiles
```

**Comment by marcelveldt** [comment]
@vk2him that's actually a great tip and  you can confirm that the files are useable by Music Assistant ?

**Comment by vk2him** [comment]
@marcelveldt  - MA sees the directories and I was able to play a few - the other errors I was getting however prevented all music to be added. I can play those files fine with the baked in media player though. I just downloaded 1.0.15 which hopefully fixes this issue, I'll report back. Thanks

**Comment by marcelveldt** [comment]
OK, that's great news. That means that SMB locations will also work as long as you mount the path to HA.

Please let me know if 1.0.15 works better for you. I tried to address all issues I could find with the file provider.

**Comment by vk2him** [comment]
@marcelveldt - 1.0.15 is much better thanks, it's found my NAS directory files and I can play them - however there are a few little bugs though. I'll create new issues now when you have a chance. thanks

**Comment by marcelveldt** [comment]
@vk2him thanks a lot for your extensive testing and creating separate issues voor each issue you've found. I will take a look at them later today.

**Comment by jeroenterheerdt** [comment]
> @marcelveldt - I have the files from my network mounted Synology NAS added to my /media directory, so this integration sees those files and can play them. Add/modify the following in your configuration.yaml Note: Replace //192.168.1.252/music with the network path to your files The above path will be mounted to the following directory (change as needed) /media/nasfiles
> 
> ```
> # For mounting Network media
> shell_command:
>   mount_nas_folder: mkdir -p /media/nasfiles;mount -t cifs -o vers=3.0,noserverino,username=nasusername,password=naspassword,domain=WORKGROUP //192.168.1.252/music /media/nasfiles
> ```

thanks, this helped me. I now run this in an automation that triggers on the `homeassistant_started` event.

**Comment by vk2him** [comment]
All - Regarding my issues where I was unable to play any music from my Network - I did some research and found that to play the music from a network location, the actual directory must be accessible from the network. The directory /media isn't accessible, however /config/www/media is, so after changing my above  snippet I can now play music from the Network NAS! This seems to have also fixed a number of errors I was getting - I'll update those issues after I check.

Of course I had to remove and reinstall the Music Assistant Integration so the database was fresh and pointing to the new locations. 

Code should be:
'''
# For mounting Network media
shell_command:
  mount_nas_folder: mkdir -p /media/nasfiles;mount -t cifs -o vers=3.0,noserverino,username=nasusername,password=naspassword,domain=WORKGROUP //192.168.1.252/music /config/www/media/nasfiles

```
As an added tip - the default media player expects it's media to be in /media , so the above will break that. An easy fix is to delete the directory /media/nasfiles and then recreate it as a soft link:

```
ln -s /config/www/media/nasfiles /media/nasfiles
```

**Comment by marcelveldt** [comment]
@vk2him that is not true but bizarre that it works. Files are accessed on local level not on smb level.
The python code from Music Assistant (which runs in the context of Home Assistant) just needs real disk access to the local file.

Your trick with mounting the smb folder is causing the side effect that we do not know if the folder is available.
It may still be mounting while MA already tries to sync. I think I managed to solve that for a large part by delaying the sync until really every aspect of HA is setup.

I think I'm going to test this approach myself by creating a fake network storage in my network

**Comment by vk2him** [comment]
> @vk2him that is not true but bizarre that it works. Files are accessed on local level not on smb level. The python code from Music Assistant (which runs in the context of Home Assistant) just needs real disk access to the local file.
> 
> Your trick with mounting the smb folder is causing the side effect that we do not know if the folder is available. It may still be mounting while MA already tries to sync. I think I managed to solve that for a large part by delaying the sync until really every aspect of HA is setup.
> 
> I think I'm going to test this approach myself by creating a fake network storage in my network

You're 100% right Marcel that the share doesn't need to be in www directory - I reverted back to the original share method and it works fine, it must have been a timing issue that you've resolved. For clarity I'll repost the correct code, and also an addition that automates this via an automation

Put this in configuration.yaml

Note:
Replace //192.168.1.252/music with the network path to your files
The above path will be mounted to the following directory (change as needed) /media/nasfiles
```
# For mounting Network media
shell_command:
  mount_nas_folder: mkdir -p /media/nasfiles;mount -t cifs -o vers=3.0,noserverino,username=nasusername,password=naspassword,domain=WORKGROUP //192.168.1.252/music /media/nasfiles
```

Put this automation code either in configuration.yaml, or if you have this in configuration.yaml ```automation: !include automations.yaml``` then put it in automations.yaml

```
- id: "mount_nas"
  alias: Mount Synology NVR Folder
  description: Mounts for use with NVR 
  trigger:
    - platform: homeassistant
      event: start
  condition: []
  action:
    - service: shell_command.mount_nas_folder
      data: {}
```

**Comment by erkr** [comment]
As there is a decent alternative, as depicted by @vk2him, I make this feature request low prio, and add the documentation label (add this solution to the documentation)

**Comment by Mariusthvdb** [comment]
posted in Discord before I saw this, sorry.
just as a variation on the shell_command above, which actually makes the dir on each restart, I do:

```
shell_command:
  mount_nas_mp3_folder: !secret mount_mp3
```

for ease and safety of sharing ;-)

and the actual command secret then is:

```
mount_mp3: mount -t cifs -o vers=3.0,noserverino,username=xxx,password=xxx,domain=WORKGROUP //192.168.1.81/Mp3 /media/music/nas/mp3
```

which also would be 1 less command to wait for in the startup sequence.

as a side note, (maybe need a dedicated issue for this, but since this currently is a hack and not yet provided by the integration itself):

interrupting building the music_assistant.db by restarting does not pickup correctly from where it left str restart. Update is going on, indicated by the spinner, and warnings are logged on missing id3 tags, proving something is happening. But the numbers dont ad up anymore in the library top menu bar for Artists, Albums, Tracks etc.

**Comment by erkr** [comment]
> interrupting building the music_assistant.db by restarting does not pickup correctly from where it left str restart

This one is worth a separate issue, thanks

**Comment by sparkydave1981** [comment]
I'm very keen to see my DLNA server files accessible in MA

**Comment by vk2him** [comment]
@sparkydave1981 - you can udr dlna server now with MA - refer to my previous note above:

https://github.com/music-assistant/hass-music-assistant/issues/71#issuecomment-1133457082
